mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6211,17 +6242,21 @@ struct GridwiseGemmV2RewritePattern
 
     int64_t GemmBBlockCopyThreadSliceLengths_GemmK;
     int64_t GemmBBlockCopyThreadSliceLengths_GemmN;
-    assert(matrix_b_source_vector_read_dim != GemmG);
-    if (matrix_b_source_vector_read_dim == GemmK) {
+    switch (matrix_b_source_vector_read_dim) {
+    case GemmK:
       GemmBBlockCopyThreadSliceLengths_GemmK = matrix_b_source_data_per_read;
       GemmBBlockCopyThreadSliceLengths_GemmN =
           GemmBBlockCopyNumberDataPerThread /
           GemmBBlockCopyThreadSliceLengths_GemmK;
-    } else if (matrix_b_source_vector_read_dim == GemmMorN) {
+      break;
+    case GemmMorN:
       GemmBBlockCopyThreadSliceLengths_GemmN = matrix_b_source_data_per_read;
       GemmBBlockCopyThreadSliceLengths_GemmK =
           GemmBBlockCopyNumberDataPerThread /
           GemmBBlockCopyThreadSliceLengths_GemmN;
+      break;
+    case GemmG:
+      llvm_unreachable("Group matrix not supported");
```

Comment:
The message should be something like : " vector loads / stores are not possible on GemmG dimension ".

Reply:
Thanks - it was hard to get context for what that case meant.